src/os/bluestore/BlueStore.cc

```diff
@@ -18299,7 +18323,34 @@ void BlueStore::_log_alerts(osd_alert_list_t& alerts)
   } else if (!spillover_alert.empty()){
     spillover_alert.clear();
   }
-
+  if (cct->_conf->bluestore_slow_op_event_counter_threshold) {
+    _trim_slow_op_event_queue(mono_clock::now());
+    if (!slow_op_event_queue.empty()) {
+      ostringstream ss;
+      uint64_t qsize = slow_op_event_queue.size();
+      ss << "got "
+         << (qsize > cct->_conf->bluestore_slow_op_event_counter_threshold ? "more than " : "") 
+         << qsize - (qsize > cct->_conf->bluestore_slow_op_event_counter_threshold) 
+         << " slow operation indication observed in last "
+         << cct->_conf->bluestore_warn_duration_for_slow_ops << " seconds";
+      alerts.emplace("BLUESTORE_SLOW_OP_ALERT", ss.str());
+    }
+  }
+  if (cct->_conf->bluestore_stalled_read_event_counter_threshold) {
+    bdev->trim_stalled_read_event_queue(mono_clock::now());
```

Comment:
IMO better have bdev->collect_alers(alerts) call hear and hide all the alert creation mechanics into BlockDevice.